tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/UsernamePasswordIntegrationTests.cs

```diff
@@ -163,25 +177,32 @@ public async Task AcquireTokenWithFederatedUsernameIncorrectPasswordAsync()
         private async Task RunHappyPathTestAsync(LabResponse labResponse)
         {
             var user = labResponse.User;
-
             SecureString securePassword = new NetworkCredential("", user.GetOrFetchPassword()).SecurePassword;
 
             var factory = new TestHttpClientFactory();
-            var msalPublicClient = PublicClientApplicationBuilder
+            var builder = PublicClientApplicationBuilder
                 .Create(labResponse.App.AppId)
-                .WithHttpClientFactory(factory)
-                .WithAuthority(_authority)
-                .Build();
+                .WithHttpClientFactory(factory);
 
+            switch (labResponse.User.AzureEnvironment)
+            {
+                case AzureEnvironment.azureusgovernment:
+                    builder.WithAuthority(labResponse.Lab.Authority + "organizations");

```

Comment:
You actually have to use `organizations` for Public AAD as well. So this is common code, i.e. you don't need a `switch`

Reply:
You can just use 

`builder.WithAuthority(labResponse.Lab.Authority, "organizations");`